app/forms/support_interface/multiple_provider_users_wizard.rb

```diff
@@ -0,0 +1,161 @@
+    def single_provider_user_form(state_store_id: nil)
+      provider_users = read_state['provider_users']
+      provider_user = if state_store_id
+                        provider_users.select { |pu| pu['state_store_id'] == state_store_id }.first
+                      else
+                        provider_users.select { |pu| pu['complete'] == false }&.first
+                      end
+
+      form = SupportInterface::CreateSingleProviderUserForm.new(
+        state_store_id: provider_user['state_store_id'],
+        provider_id: provider_id,
+        first_name: provider_user['first_name'],
+        last_name: provider_user['last_name'],
+        email_address: provider_user['email_address'],
+      )
+
+      if provider_user['permissions']
+        form.provider_permissions = { provider_permission: provider_user['permissions'].symbolize_keys }
+      end
+
+      form
+    end
+
+  private
+
+    def prepare_provider_users_for_state_store
+      provider_user_array_hash = provider_users
+        .split("\r\n")
+        .map { |row| row.split(/[\t,]+/) }
+        .map do |provider_user|
+        {
+          state_store_id: SecureRandom.uuid,
+          first_name: provider_user[0],
+          last_name: provider_user[1],
+          email_address: provider_user[2],
+          complete: false,
+        }
+      end
+
+      {
+        provider_users: provider_user_array_hash,
+      }.to_json
+    end
+
+    def prepare_provider_user_for_state_store(single_provider_user_form)
+      f = single_provider_user_form
+      state = read_state
+
+      updated_user_details = {
+        state_store_id: f.state_store_id,
+        first_name: f.first_name,
+        last_name: f.last_name,
+        email_address: f.email_address,
+        permissions: f.permission_form.provider_permission,
+        complete: true,
+      }
+      index = state['provider_users'].index { |pu| [f.state_store_id].include?(pu['state_store_id']) }
+
+      state['provider_users'].reject! { |pu| pu['state_store_id'] == f.state_store_id }
+      state['provider_users'][index, 0] = updated_user_details
+
+      clear_state!
+      state.to_json
+    end
+
+    def prepare_data_for_db_save
+      provider_users = read_state['provider_users']
+      forms = provider_users.map do |provider_user|
+        SupportInterface::CreateSingleProviderUserForm.new(
+          provider_id: provider_id,
+          first_name: provider_user['first_name'],
+          last_name: provider_user['last_name'],
+          email_address: provider_user['email_address'],
+          provider_permissions: {
+            provider_permission: provider_user['permissions'],
+          },
+        )
+      end
+
+      @prepared_db_data = forms.map do |form|
+        provider_user = form.build
+
+        SaveAndInviteProviderUser.new(
+          form: form,
+          save_service: SaveProviderUser.new(
+            provider_user: provider_user,
+            provider_permissions: [form.provider_permissions],
```

Comment:
It seems we aren't saving the provider permissions (when I navigate away to try and edit the user I just added) are we passing in the permissions correctly to `CreateSingleProviderUserForm`?

Reply:
Nothing is persisted to the db until after the 'Check' page ... Not sure I understand can we discuss?

Reply:
This issue is fixed now